prefect_dask/task_runners.py

```diff
@@ -213,14 +213,17 @@ async def submit(
         if "task_run" in call_kwargs:
             task_run = call_kwargs["task_run"]
             flow_run = FlowRunContext.get().flow_run
+            # Dask displays the text up to the first '-' as the name; the task run key
+            # should include the task run name for readability in the dask console.
+            # The flow run count is included for cases where keys will match
+            # if the task run failed and is running again for a retried flow run,
+            # resulting in a retrieval from the Dask cache.
```

Comment:
Note: We do _not_ want this to result in retrieval of the failed state from the Dask cache which is why we give it a unique key per flow run retry.